Remove unnecessary copy when checking if a page is running an extension

bnham opened the pull request:
#### 35745daf8d3505bc1f273e3fa5a3e4229dffa70d
<pre>
Remove unnecessary copy when checking if a page is running an extension
<a href="https://bugs.webkit.org/show_bug.cgi?id=270321">https://bugs.webkit.org/show_bug.cgi?id=270321</a>
<a href="https://rdar.apple.com/problem/123858678">rdar://problem/123858678</a>

Reviewed by Timothy Hatcher.

For every navigation decision, we check whether the associated page is running a web extension by
calling `page.cocoaView().get().configuration._requiredWebExtensionBaseURL`. Since configuration is
marked as a `copy` property, this causes an allocation and copy every time it&apos;s called. We should
avoid the copy since it&apos;s not useful.

I have another use case for `_requiredWebExtensionBaseURL` outside of WebKit, so I put it in
WKWebViewPrivate.h rather than WKWebViewInternal.h.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _requiredWebExtensionBaseURL]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/Cocoa/NavigationState.mm:
(WebKit::isUnsupportedWebExtensionNavigation):

Canonical link: <a href="https://commits.webkit.org/275538@main">https://commits.webkit.org/275538@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1a9b6e93b576380750d432e6d8e8f9b495dafba4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/42096 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/21114 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/44490 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44683 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/38209 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/24295 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/18442 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/34858 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42670 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/18035 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/36241 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/15791 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/15704 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/46116 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/38277 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37601 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/41507 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16912 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13897 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/40064 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/18531 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9430 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18591 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/18176 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

xeenon commented on Source/WebKit/UIProcess/Cocoa/NavigationState.mm:
```diff
@@ -514,7 +514,7 @@ static void tryInterceptNavigation(Ref<API::NavigationAction>&& navigationAction
 static bool isUnsupportedWebExtensionNavigation(API::NavigationAction& navigationAction, WebPageProxy& page)
 {
-    auto *requiredBaseURL = page.cocoaView().get().configuration._requiredWebExtensionBaseURL;
+    auto *requiredBaseURL = [page.cocoaView() _requiredWebExtensionBaseURL];
```
I still prefer dot-syntax with `.get()`, but this is fine.

bnham committed:
Remove unnecessary copy when checking if a page is running an extension
https://bugs.webkit.org/show_bug.cgi?id=270321
rdar://problem/123858678

Reviewed by Timothy Hatcher.

For every navigation decision, we check whether the associated page is running a web extension by
calling `page.cocoaView().get().configuration._requiredWebExtensionBaseURL`. Since configuration is
marked as a `copy` property, this causes an allocation and copy every time it's called. We should
avoid the copy since it's not useful.

I have another use case for `_requiredWebExtensionBaseURL` outside of WebKit, so I put it in
WKWebViewPrivate.h rather than WKWebViewInternal.h.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _requiredWebExtensionBaseURL]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/Cocoa/NavigationState.mm:
(WebKit::isUnsupportedWebExtensionNavigation):

Canonical link: https://commits.webkit.org/275538@main
